packages/framework/src/Model/Product/Search/ProductSearchRepository.php

```diff
@@ -42,58 +38,13 @@ public function filterBySearchText(QueryBuilder $productQueryBuilder, $searchTex
      */
     public function addRelevance(QueryBuilder $productQueryBuilder, $searchText)
     {
-        $productQueryBuilder->addSelect('
-            CASE
-                WHEN (
-                    NORMALIZE(pt.name) LIKE NORMALIZE(:searchTextLikeWithWildcardOnLeftAndSpaceAndWildcardOnRight)
-                    OR
-                    NORMALIZE(pt.name) LIKE NORMALIZE(:searchTextLikeWithWildcardOnLeft)
-                ) THEN 1
-                WHEN TSQUERY(pt.nameTsvector, :searchTextTsqueryAnd) = TRUE THEN 2
-                WHEN TSQUERY(p.catnumTsvector, :searchTextTsqueryOr) = TRUE THEN 3
-                WHEN TSQUERY(p.partnoTsvector, :searchTextTsqueryOr) = TRUE THEN 4
-                WHEN NORMALIZE(pt.name) LIKE NORMALIZE(:searchTextLikeWithWildcardsOnBothSides) THEN 5
-                WHEN TSQUERY(pt.nameTsvector, :searchTextTsqueryAndWithPrefixMatchForLastWord) = TRUE THEN 6
-                WHEN TSQUERY(pd.descriptionTsvector, :searchTextTsqueryAnd) = TRUE THEN 7
-                WHEN TSQUERY(pt.nameTsvector, :searchTextTsqueryOr) = TRUE THEN 8
-                WHEN TSQUERY(pt.nameTsvector, :searchTextTsqueryOrWithPrefixMatchForLastWord) = TRUE THEN 9
-                ELSE 10
-            END AS HIDDEN relevance
-        ');
-
-        $productQueryBuilder->setParameter(
-            'searchTextLikeWithWildcardsOnBothSides',
-            DatabaseSearching::getFullTextLikeSearchString($searchText)
-        );
-
-        $productQueryBuilder->setParameter(
-            'searchTextLikeWithWildcardOnLeft',
-            '%' . DatabaseSearching::getLikeSearchString($searchText)
-        );
-
-        $productQueryBuilder->setParameter(
-            'searchTextLikeWithWildcardOnLeftAndSpaceAndWildcardOnRight',
-            '%' . DatabaseSearching::getLikeSearchString($searchText) . ' %'
-        );
-
-        $productQueryBuilder->setParameter(
-            'searchTextTsqueryAnd',
-            $this->tsqueryFactory->getTsqueryWithAndConditions($searchText)
-        );
+        $domainId = $productQueryBuilder->getParameter('domainId')->getValue();
+        $productIds = $this->client->search($domainId, $searchText);
 
-        $productQueryBuilder->setParameter(
-            'searchTextTsqueryAndWithPrefixMatchForLastWord',
-            $this->tsqueryFactory->getTsqueryWithAndConditionsAndPrefixMatchForLastWord($searchText)
-        );
-
-        $productQueryBuilder->setParameter(
-            'searchTextTsqueryOr',
-            $this->tsqueryFactory->getTsqueryWithOrConditions($searchText)
-        );
-
-        $productQueryBuilder->setParameter(
-            'searchTextTsqueryOrWithPrefixMatchForLastWord',
-            $this->tsqueryFactory->getTsqueryWithOrConditionsAndPrefixMatchForLastWord($searchText)
-        );
+        if (count($productIds) > 0) {
```

Comment:
I see a kind of inconsistency here :/ A few lines above there is only `(count($productIds))` and here is `(count($productIds) > 0)`